module/documents/actor/actor.mjs

```diff
@@ -1359,191 +1359,208 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
+
+      let prof = process.prof ?? relevant?.prof;
+      if ( prof && hasRemarkableAthlete ) {
+        const validAbilities = CONFIG.DND5E.characterFlags.remarkableAthlete.abilities;
+        if ( validAbilities.includes(abilityId) && (prof.multiplier <= .5) && (prof.rounding === "down") ) {
+          prof = prof.clone({ multiplier: .5, roundDown: false });
+        } else if ( !validAbilities.includes(abilityId) && (prof.multiplier === .5) && (prof.rounding === "up") ) {
+          if ( this.getFlag("dnd5e", "jackOfAllTrades") ) prof = prof.clone({ roundDown: true });
+          else prof = prof.clone({ multiplier: 0 });
+        }
+      }
 
-    // Add exhaustion reduction
-    this.addRollExhaustion(parts, data);
+      let { parts, data } = CONFIG.Dice.BasicRoll.constructParts({
+        mod: ability?.mod,
+        prof: prof?.hasProficiency ? prof.term : null,
+        [`${config[type]}Bonus`]: relevant?.bonuses?.check,
+        extraBonus: process.bonus,
+        [`${abilityId}CheckBonus`]: ability?.bonuses?.check,
+        [`${type}Bonus`]: this.system.bonuses?.abilities?.[type],
+        abilityCheckBonus: this.system.bonuses?.abilities?.check,
+        situational: config.data?.situational
+      }, { ...rollData });
+      const options = config.options ?? {};
+
+      if ( index === 0 ) {
+        if ( initialRoll?.data ) data = { ...data, ...initialRoll.data };
+        if ( initialRoll?.parts ) parts.unshift(...initialRoll.parts);
+        if ( initialRoll?.options ) foundry.utils.mergeObject(options, initialRoll.options);
+      }
 
-    // Reliable Talent applies to any tool check we have full or better proficiency in
-    const reliableTalent = (prof?.multiplier >= 1 && this.getFlag("dnd5e", "reliableTalent"));
+      // Add exhaustion reduction
+      this.addRollExhaustion(parts, data);
 
-    const flavor = game.i18n.format("DND5E.ToolPromptTitle", {tool: Trait.keyLabel(toolId, {trait: "tool"}) ?? ""});
-    const rollData = foundry.utils.mergeObject({
-      data, flavor,
-      title: `${flavor}: ${this.name}`,
-      chooseModifier: true,
+      config.parts = parts;
+      config.data = data;
+      config.data.abilityId = abilityId;
+      config.options = options;
+    };
+
+    const rollConfig = foundry.utils.mergeObject({
+      ability: defaultAbility,
       halflingLucky: this.getFlag("dnd5e", "halflingLucky"),
-      reliableTalent,
-      messageData: {
-        speaker: options.speaker || ChatMessage.implementation.getSpeaker({actor: this}),
-        "flags.dnd5e.roll": {type: "tool", toolId}
+      reliableTalent: (relevant?.value >= 1) && this.getFlag("dnd5e", "reliableTalent"),
+      rolls: [{
+        options: {
+          advantage: relevant?.roll.mode === CONFIG.Dice.D20Roll.ADV_MODE.ADVANTAGE,
+          disadvantage: relevant?.roll.mode === CONFIG.Dice.D20Roll.ADV_MODE.DISADVANTAGE,
+          maximum: relevant?.roll.max,
+          minimum: relevant?.roll.min
+        }
+      }]
```

Comment:
This `rolls` array we define here will be clobbered by `config.rolls` if it is also defined.